drivers/wifi/eswifi/eswifi_core.c

```diff
@@ -378,6 +378,8 @@ static int eswifi_get_mac_addr(struct eswifi_dev *eswifi, uint8_t addr[6])
 	return 0;
 }
 
+int eswifi_socket_create(int family, int type, int proto);
+
```

Comment:
It will be much safer to declare that function in `eswifi.h` file, so that compiler/linker will detect if there is difference between declaration and definition.